README.md

```diff
@@ -1,16 +1,63 @@
 # Exercises
 
-__NOTE:__ DO NOT USE THE BUILT IN METHODS WHEN ASKED TO MAKE THEM! 
+__NOTE:__ DO NOT USE THE BUILT IN METHODS WHEN ASKED TO MAKE THEM!
 
 1. Using Array.prototype add a new method `myForEach` that mimics the behaivor of the built in `forEach`.
 
-2. Use your `myForEach` to add `myMap` to the Array prototype. `myMap` should behave the same as regular `map`. 
+Array.prototype.myForEach = function(callback) {
+for(let i = 0; i < names.length; i++) {
+   callback(this[i]);
+}
+}
+Array(1,2,3)
 
-3. Write a 'myFilter` that behave's the same as `filter`. 
+2. Use your `myForEach` to add `myMap` to the Array prototype. `myMap` should behave the same as regular `map`.
+
+Array.prototype.myMap = function(callback) {
+  let output = []
+for(let i = 0; i < arr.length; i++) {
+    callback(output.push(this[i]));
+}return output
+}
+
+Array(1,2,3)
+
+//I don't understand what map does, I can parrot that it takes an array, performs a function, and returns a new array; but this does not equate understanding.
+
+3. Write a 'myFilter' that behaves the same as filter.
+
+Array.prototype.myFilter = function(callback){
+  let output = []
+  for(let i = 0; i < arr.length; i++){
+    if(arr[i]){
+      output.push(i)
+    }
+  }
+}
+
+4. Write a `myEvery`.
+Array.prototype.myEvery = function(callback){
+  for(i = 0; i < arr.length; i++){
+    if( arr[i] === arr[j]){
+      return true
+    }
+  }
+}
+//don't understand the assignment
+
+5. Write a `myReduce`. It should use the first element in the array if none is given.
+
+Array.prototype.myReduce = function(callback){
```

Comment:
so, this one is really tricky, and is all about what reduce is supposed to be doing - that is, synthesizing each element in a way that the callback dictates.